lib/language_variants_filter.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const mwUtil = require('./mwUtil');
+
+module.exports = (hyper, req, next, options, specInfo) => {
+    const rp = req.params;
+    const acceptLanguage = req.headers && req.headers['accept-language'];
+    if (!acceptLanguage || mwUtil.isNoCacheRequest(req)) {
+        delete req.headers['accept-language'];
+        return next(hyper, req);
+    }
+
+    const revTableURI = [rp.domain, 'sys', 'page_revisions', 'page', rp.title];
+    if (rp.revision) {
+        revTableURI.push(`${rp.revision}`);
+    }
+    return hyper.get({ uri: new URI(revTableURI) })
+    .then((res) => {
+        const revision = res.body.items[0];
+        return mwUtil.shouldConvertLangVariant(hyper, req, revision.page_language, acceptLanguage)
+        .then((shouldConvert) => {
+            if (!shouldConvert) {
+                delete req.headers['accept-language'];
+                return next(hyper, req);
+            }
+
+            if (/\/page\/html\//.test(specInfo.path)) {
+                // It's HTML, hit Parsoid for conversion
+                return next(hyper, req)
+                .then(html => hyper.post({
```

Comment:
All this block looks like it belongs in the Parsoid module rather than here since we keep all Parsoid-related interfaces and logic there.

Reply:
It's basically just making a request to the parsoid module here. We can probably move a bit more logic in there, but not too much..